src/data/equiv/mul_add.lean

```diff
@@ -418,6 +418,19 @@ lemma Pi_congr_right_trans {η : Type*}
   (es : ∀ j, Ms j ≃* Ns j) (fs : ∀ j, Ns j ≃* Ps j) :
   (Pi_congr_right es).trans (Pi_congr_right fs) = (Pi_congr_right $ λ i, (es i).trans (fs i)) := rfl
 
+
+/-- A family indexed by a unique type is equivalent to the element at the default value -/
+@[to_additive add_equiv.Pi_unique "A family indexed by a unique type is equivalent to the
+element at the default value"]
+def Pi_unique
+  {η : Type*} (M : η → Type*) [Π (j : η), mul_one_class (M j)] [unique η] :
+  (Π (j : η), M j) ≃* M default :=
```

Comment:
I would imagine this turns out more useful if you replace `[unique η]` with `[subsingleton η] (i : η)`, as this gives definitional control of the index

Reply:
This is how I'm using it: https://github.com/leanprover-community/mathlib/pull/11778

So there I just need `default`.  But of course I could pass `default` to the lemma.

Reply:
Or I'll simply provide both variants, for maximal usefulness.

Reply:
Ok, there is now `Pi_subsingleton` and `Pi_unique`.

Reply:
> This is how I'm using it: https://github.com/leanprover-community/mathlib/pull/11778

I bet you wouldn't need the `convert` there if you used the `subsingleton` version instead of the `unique` version - so I'd be inclined to remove the `unique` version, it's likely just harder to use.

Reply:
Your bet is correct! I’ll remove the `unique` variants.

Reply:
Note this is being reverted back to the `Unique` version in https://github.com/leanprover-community/mathlib4/pull/8249